server/service/apple_mdm.go

```diff
@@ -721,6 +746,86 @@ func (svc *Service) DeleteMDMAppleConfigProfile(ctx context.Context, profileUUID
 	return nil
 }
 
+func (svc *Service) DeleteMDMAppleDeclaration(ctx context.Context, declUUID string) error {
```

Comment:
We could probably find a way handle the service layer without three different functions for deletions. For now, we're duplicating a bit of code.